spec/models/data_file_spec.rb

```diff
@@ -262,53 +262,100 @@
       end
     end
 
-    describe '#set_deleted_from_parent' do
-      it { is_expected.to respond_to(:set_deleted_from_parent) }
+    describe '.move_to_trashbin' do
+      it { is_expected.to respond_to(:move_to_trashbin) }
 
-      context 'being restored' do
-        subject { FactoryBot.create(:data_file, :deleted, :with_parent) }
-        it {
-          subject.deleted_from_parent_id = subject.parent_id
-          subject.parent_id = nil
-          subject.save
-
-          expect(subject.parent_id).to be_nil
-          expect(subject.parent).to be_nil
-          expect(subject.deleted_from_parent_id).not_to be_nil
-          expect(subject.deleted_from_parent).not_to be_nil
-          original_parent = subject.deleted_from_parent
-
-          subject.is_deleted = false
-          expect(subject.is_deleted_was).to be_truthy
-          subject.set_deleted_from_parent
-
-          expect(subject.parent_id).not_to be_nil
-          expect(subject.parent).not_to be_nil
-          expect(subject.deleted_from_parent_id).to be_nil
-          expect(subject.deleted_from_parent).to be_nil
-          expect(subject.parent).to eq(original_parent)
-        }
-      end
+      subject { child_file }
+      it {
+        expect(subject.parent_id).not_to be_nil
+        expect(subject.parent).not_to be_nil
+        original_parent = subject.parent
```

Comment:
You could set `original_parent` as a let variable and then, instead of having `expect(subject.parent).not_to be_nil`, use `expect(original_parent).not_to be_nil`